comp/networkpath/npscheduler/npschedulerimpl/npscheduler.go

```diff
@@ -7,31 +7,324 @@
 package npschedulerimpl
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
 	model "github.com/DataDog/agent-payload/v5/process"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
 	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
+	"github.com/DataDog/datadog-agent/comp/networkpath/npscheduler/npschedulerimpl/common"
+	"github.com/DataDog/datadog-agent/comp/networkpath/npscheduler/npschedulerimpl/pathteststore"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/networkpath/metricsender"
+	"github.com/DataDog/datadog-agent/pkg/networkpath/payload"
+	"github.com/DataDog/datadog-agent/pkg/networkpath/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/networkpath/traceroute"
+	"github.com/DataDog/datadog-agent/pkg/process/statsd"
+	ddgostatsd "github.com/DataDog/datadog-go/v5/statsd"
+	"go.uber.org/atomic"
 )
 
 type npSchedulerImpl struct {
-	epForwarder eventplatform.Component
-	enabled     bool
+	enabled bool
+
+	// Deps
+	epForwarder  eventplatform.Forwarder
+	logger       log.Component
+	metricSender metricsender.MetricSender
+	statsdClient ddgostatsd.ClientInterface
+
+	// Counters
+	receivedPathtestCount    *atomic.Uint64
+	processedTracerouteCount *atomic.Uint64
+
+	// Pathtest store
+	pathtestStore       *pathteststore.PathtestStore
+	pathtestInputChan   chan *common.Pathtest
+	pathtestProcessChan chan *pathteststore.PathtestContext
+
+	// Scheduling related
+	running       bool
+	workers       int
+	stopChan      chan struct{}
+	flushLoopDone chan struct{}
+	runDone       chan struct{}
+	flushInterval time.Duration
+
+	// structures needed to ease mocking/testing
+	TimeNowFn func() time.Time
+	// TODO: instead of mocking traceroute via function replacement like this
+	//       we should ideally create a fake/mock traceroute instance that can be passed/injected in NpScheduler
+	runTraceroute func(cfg traceroute.Config) (payload.NetworkPath, error)
 }
 
-func (s *npSchedulerImpl) ScheduleConns(_ []*model.Connection) {
+func newNoopNpSchedulerImpl() *npSchedulerImpl {
+	return &npSchedulerImpl{enabled: false}
+}
+
+func newNpSchedulerImpl(epForwarder eventplatform.Forwarder, logger log.Component, sysprobeYamlConfig config.Reader) *npSchedulerImpl {
+	workers := sysprobeYamlConfig.GetInt("network_path.workers")
```

Comment:
Are there sane defaults for some of these configs?

Reply:
I think they are reasonable for now, but might need to be tweak once we have more data. We will have more data by:
- deploying Network Path using SNMP Conns in real environment (feature in this PR)
- once we better optimised the traceroute implementation (this is using most of the CPU resources, but should/will be addressed separate, not directly related to this PR)

Reply:
This would mean all of these would have to be set in the config when `network_path.enabled` is enabled. Usually we do define some sane-ish defaults here, just so that if these are not defined we don't see panics at the very least.

Reply:
@hmahmood Hum, as part of this PR, there are already defaults values set in `pkg/config/setup/system_probe.go` 

<img width="756" alt="image" src="https://github.com/DataDog/datadog-agent/assets/49917914/74b0e22c-f7f9-4de4-9770-c6dc2c920475">


Maybe I'm misunderstanding what you mean.

Reply:
That is what I was looking for :+1: